wordpress/wp-content/plugins/memberful-wp/src/global_marketing.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+if ( ! defined( 'MEMBERFUL_PARAGRAPH_COUNT' ) ) {
+	define( 'MEMBERFUL_PARAGRAPH_COUNT', 2 );
+}
+
+add_filter( 'memberful_wp_protect_content', 'memberful_apply_global_marketing_content_filter', 15, 1 );
+
+/**
+ * Filter the paywall to return a "teaser".
+ *
+ * @param string $memberful_marketing_content
+ *
+ * @return string concat of teaser and memberful marketing content
+ */
+function memberful_apply_global_marketing_content_filter( $memberful_marketing_content ) {
+	global $post;
+	$override = get_option( 'memberful_global_marketing_override' );
+
+	if ( ! $override && ! empty( $memberful_marketing_content ) ) {
+		return $memberful_marketing_content;
+	}
+
+	$global_marketing_content         = get_option( 'memberful_global_marketing_content' );
+	$wrapped_global_marketing_content = "<div class='memberful-global-marketting-content'>$global_marketing_content</div>";
+
+	// Prevent endless loop trap
+	remove_action( 'the_content', 'memberful_wp_protect_content', -10 );
+
+	$original_content = apply_filters( 'the_content', $post->post_content );
+
+	// re-add the action for follow-on call
+	add_action( 'the_content', 'memberful_wp_protect_content', -10 );
+
+	$offset = 0;
+	for ( $i = 0; $i < MEMBERFUL_PARAGRAPH_COUNT; $i++ ) {
+		$offset = strpos( $original_content, '</p>', $offset ) + 5;
```

Comment:
Does this assume all content for posts will be in <p> tags?

I'm wondering how this would handle a post that is made up of content in a variety of other tags like `div`, `span`, `blockquote`, `video` etc. Or a post that has a lot of other tags before a couple of <p> tags at the end. Would that mean the offset includes all that other content in the teaser?

Reply:
@northerner Correct. That is a deficiency in the way this feature currently functions, it really only works for long-form content. We could open up the project scope and requirements to allow the user to specify their tags -- but that would add some complexity and cost.

Alternately, we could fall back to using wp_excerpt, if that's preferable.

Reply:
I think that using `<p>` tags is OK for now. We can add support for other tags based on user feedback.